File: tests/agent_workloads/adapters/process_batch.py
import subprocess
import sys
from pathlib import Path


if len(sys.argv) != 2:
    raise SystemExit("usage: process_batch INPUT_FILE")

for line_number, line in enumerate(Path(sys.argv[1]).read_text().splitlines()[1:], 2):
    if not line:
        continue
    fields = line.split("\t")
    if len(fields) != 4:
        raise SystemExit(f"bad process row {line_number}")
    label, program, argument, timeout_text = fields
    try:
        completed = subprocess.run(
            [program, argument],
            capture_output=True,
            text=True,
            timeout=int(timeout_text) / 1000,
            check=False,
        )
    except subprocess.TimeoutExpired:
        print(f"{label}|timeout")
    else:
        print(f"{label}|exit={completed.returncode}|stdout={completed.stdout.strip()}")

File: tests/agent_workloads/adapters/process_batch.mjs
import { spawn } from "node:child_process";
import { readFileSync } from "node:fs";

if (process.argv.length !== 3) {
  throw new Error("usage: process_batch INPUT_FILE");
}

function run(program, argument, timeoutMs) {
  return new Promise((resolve, reject) => {
    const child = spawn(program, [argument], { stdio: ["ignore", "pipe", "pipe"] });
    let output = "";
    let timedOut = false;
    child.stdout.setEncoding("utf8");
    child.stdout.on("data", (chunk) => {
      output += chunk;
    });
    child.on("error", reject);
    const timer = setTimeout(() => {
      timedOut = true;
      child.kill("SIGKILL");
    }, timeoutMs);
    child.on("close", (code) => {
      clearTimeout(timer);
      resolve({ code, output: output.trim(), timedOut });
    });
  });
}

const lines = readFileSync(process.argv[2], "utf8").split(/\r?\n/);
for (let index = 1; index < lines.length; index += 1) {
  const line = lines[index];
  if (!line) continue;
  const fields = line.split("\t");
  if (fields.length !== 4) {
    throw new Error(`bad process row ${index + 1}`);
  }
  const [label, program, argument, timeoutText] = fields;
  const result = await run(program, argument, Number.parseInt(timeoutText, 10));
  if (result.timedOut) {
    console.log(`${label}|timeout`);
  } else {
    console.log(`${label}|exit=${result.code}|stdout=${result.output}`);
  }
}
